Subject: Make Amarok2 play ogg with gstreamer
From: Mandriva phonon package
Signed-Off-By: Lubos Lunak
Bug: 438718
Relates: kde4-amarok/ogg-mimetypes.diff
Patch-upstream: no

I found this in Mandriva phonon package, I'm not quite sure what it does
or why does it do it, but it looks ok and helps.

Index: gstreamer/backend.cpp
===================================================================
--- gstreamer/backend.cpp	(revision 862019)
+++ gstreamer/backend.cpp	(working copy)
@@ -226,6 +227,15 @@
         }
     }
     g_list_free(factoryList);
+    if (availableMimeTypes.contains("audio/x-vorbis")
+        && availableMimeTypes.contains("application/x-ogm-audio")) {
+        if (!availableMimeTypes.contains("audio/x-vorbis+ogg"))
+            availableMimeTypes.append("audio/x-vorbis+ogg");
+        if (!availableMimeTypes.contains("application/ogg"))  /* *.ogg */
+            availableMimeTypes.append("application/ogg");
+        if (!availableMimeTypes.contains("audio/ogg")) /* *.oga */
+            availableMimeTypes.append("audio/ogg");
+    }
     availableMimeTypes.sort();
     return availableMimeTypes;
 }
